ompi/mca/fcoll/static/fcoll_static_file_read_all.c

```diff
@@ -482,15 +482,15 @@ mca_fcoll_static_file_read_all (mca_io_ompio_file_t *fh,
             }
         }
 
-        if (local_cycles > index) {
-            if ((index == local_cycles-1) && (max_data % bytes_per_cycle)) {
-                bytes_to_read_in_cycle = max_data % bytes_per_cycle;
+        if (index < local_cycles ) {
+            if ((index == local_cycles-1) && (max_data % (bytes_per_cycle/fh->f_procs_per_group))) {
+                bytes_to_read_in_cycle = max_data - position;
```

Comment:
The fix looks good to me. Just a small suggestion,
Use total_bytes_read instead of position? I am verifying the patch now locally

Thanks